app/controllers/concerns/scopes_per_action.rb

```diff
@@ -0,0 +1,37 @@
+# This module will hold a registry of scope directives that should be
+# part of the query per specific action in a controller. This is required by
+# plugins that add *_to_many relations to foreman core models like
+# `Host::Managed` and want to add some includes or eager_load directives to
+# the query.
+
+module ScopesPerAction
+  extend ActiveSupport::Concern
+
+  # returns a scope that includes all directives registered via add_scope_for
+  def action_scope_for(action, base_scope)
+    scope = base_scope
+    self.class.scopes_for(action).each do |scope_func|
+      scope = scope_func.call(scope) || scope
```

Comment:
I am chaining the scopes, here is an example:
In puppet plugin:
``` ruby
HostsController.add_scope_for(:index) { |base_scope| base_scope.eager_load(:environment) }
```
In provisioning plugin:

``` ruby
HostsController.add_scope_for(:index) { |base_scope| base_scope.eager_load(:operatingsystem) }
```

I want to create a scope that will contain both operations:

``` ruby
base_scope = Hosts.authorized
actual_scope = base_scope.eager_load(environment).eager_load(operatingsystem)
```

I achieve this by calling each block, and passing it the result of the previous block.